packages/account/src/Sections/Verification/ProofOfAddress/proof-of-address-container.jsx

```diff
@@ -10,51 +10,73 @@ import Verified from 'Components/poa-verified';
 import NotRequired from 'Components/poa-not-required';
 import PoaStatusCodes from 'Components/poa-status-codes';
 import ProofOfAddressForm from './proof-of-address-form.jsx';
+import { populateVerificationStatus } from '../Helpers/verification';
 
-const ProofOfAddressContainer = ({ is_mx_mlt, refreshNotifications }) => {
+const ProofOfAddressContainer = ({ is_mx_mlt, is_switching, refreshNotifications }) => {
     const [is_loading, setIsLoading] = React.useState(true);
     const [authentication_status, setAuthenticationStatus] = useStateCallback({
-        allow_resubmission: false,
+        allow_document_upload: false,
+        allow_poi_resubmission: false,
         needs_poi: false,
+        needs_poa: false,
         has_poi: false,
         resubmit_poa: false,
-        submitted_poa: false,
+        has_submitted_poa: false,
         document_status: null,
     });
 
     // componentDidMount hook
     React.useEffect(() => {
-        WS.authorized.getAccountStatus().then(response => {
-            const { get_account_status } = response;
-            const { document, needs_verification } = get_account_status.authentication;
-            const needs_poi = needs_verification?.includes('identity');
-            const allow_resubmission = needs_verification?.includes('document');
-            const submitted_poa = document.status === PoaStatusCodes.pending && !allow_resubmission;
-            const document_status = document.status;
-            setAuthenticationStatus(
-                { ...authentication_status, ...{ allow_resubmission, needs_poi, document_status, submitted_poa } },
-                () => {
-                    setIsLoading(false);
-                    refreshNotifications();
-                }
-            );
-        });
-    }, []); // eslint-disable-line react-hooks/exhaustive-deps
+        if (!is_switching) {
+            WS.authorized.getAccountStatus().then(response => {
+                const { get_account_status } = response;
+                const { allow_document_upload, allow_poi_resubmission, needs_poi, needs_poa, document_status } =
+                    populateVerificationStatus(get_account_status);
+                const has_submitted_poa = document_status === PoaStatusCodes.pending && !allow_poi_resubmission;
+
+                setAuthenticationStatus(
+                    {
+                        ...authentication_status,
+                        ...{
+                            allow_document_upload,
+                            allow_poi_resubmission,
+                            needs_poi,
+                            needs_poa,
+                            document_status,
+                            has_submitted_poa,
+                        },
+                    },
+                    () => {
+                        setIsLoading(false);
+                        refreshNotifications();
+                    }
+                );
+            });
+        }
+    }, [is_switching]);
```

Comment:
Added `is_switching` to avoid function to be called immediately and cause error when account is switching.